### PR TITLE
Refactor distro detection

### PR DESCRIPTION
**Pull request by Revalioli**
The ansible fact `ansible_distribution` work fine to detect if a Seapath distribution is Debian or CentOS, but doesn't work well for Yocto Seapath.
With Yocto, `ansible_distribution` will be set to the value of DISTRO_NAME. As it is usual to have multiple distro configurations in a Yocto project, with different DISTRO_NAME values, `ansible_distribution` wouldn't have a deterministic value for every Yocto Seapath build.

With yocto scarthgap, the `/etc/os-release` file has a field CPE_NAME which will always contains "openembedded".
It can then be used as a reliable way to detect a Yocto Seapath distribution.

This PR propose a new role, detect_seapath_distro, that will set a fact `seapath_distro` with a value of either Yocto, Debian or CentOS. To detect the correct distro, is uses `ansible_distribution*` facts and the /etc/os-release file.
The new fact is then used everywhere a task needs to know the seapath distribution of the host.

### CODE REVIEW

**Comment by Revalioli** [comment]
Note: currently only the last two commits are for this PR. Other commits belongs to https://github.com/seapath/ansible/pull/643, and will be removed when https://github.com/seapath/ansible/pull/643 is merged.

File: playbooks/seapath_setup_main.yaml
Comment (insatomcat) [inline playbooks/seapath_setup_main.yaml:20]: import_playbook does not take "when" into account

File: roles/detect_seapath_distro/tasks/main.yaml
Comment (dupremathieu) [inline roles/detect_seapath_distro/tasks/main.yaml:22]: I am not sure we can rely on CPE_NAME because we don't know how CPEs are defined.

Comment (Revalioli) [reply roles/detect_seapath_distro/tasks/main.yaml:22]: There is an official CPE_NAME dictionary that is used to define CVE if I understood well, but nothing says explicitly that non-official CPE shouldn't be used.
Also `cpe:/o:openembedded` isn't even in this dictionary.

This pattern for the CPE_NAME variable is hardcoded this way in poky, so it probably won't change for scarthgap.
The goal here is only to use what is hardcoded in poky, and not to set the CPE itself, maybe we can use it for now, and if in the future its CPE_NAME changes in poky or Seapath needs its own CPE, we would then adapt the role ?

File: playbooks/seapath_setup_main.yaml
Comment (insatomcat) [reply playbooks/seapath_setup_main.yaml:20]: yes, sorry, I wasn't clear but that's why I meant

Comment (Revalioli) [reply playbooks/seapath_setup_main.yaml:20]: What would you propose instead ?
`import_playbook` cannot be used inside a play, so not possible to surround it with a block or another statement on which to put the `when`.
The other way would be to put multiple `when` directly in the playbooks be we would then still get multiple "skipping" messages, and it would make a lot of repetitive `when` conditionals

Comment (insatomcat) [reply playbooks/seapath_setup_main.yaml:20]: I would probably go with having separate "main" playbooks for each flavor, or leave it to the user to create its own playbook.
But this is future work and is not related to this PR.

**Comment by Revalioli** [comment]
PR is now rebased on the latest debiancentos commits.

Note that the change from `ansible_distribution` to `seapath_distro` variable was done everywhere in the repo except for specific Debian playbooks and templates (examples: `debian_physical_machine` role, `seapath_setup_prerequisdebian.yaml` playbook, groups and users configuration template files). Only playbooks and roles that are explicitly made for Debian or CentOS are concerned.

If you think `ansible_distribution` should be changed anyway don't hesitate to comment

File: roles/add_livemigration_user/README.md
Comment (eroussy) [inline roles/add_livemigration_user/README.md:7]: Why do you need this role as a dependency ?
There is nothing in the role that uses distro specific variables

**Comment by insatomcat** [comment]
Can you create the meta/main.yml file in this new role so that it's uploadable to galaxy? thx.

File: roles/ceph_prepare_installation/README.md
Comment (eroussy) [inline roles/ceph_prepare_installation/README.md:7]: same, why do you need this dependency ?

**Review by eroussy — CHANGES_REQUESTED** [review]

**Comment by Revalioli** [comment]
> Can you create the meta/main.yml file in this new role so that it's uploadable to galaxy? thx.

@insatomcat Thanks, done
Note that I kept the `platforms` section to match other roles, but it seems that recent version of Ansible Galaxy doesn't use this metadata anymore, see https://github.com/ansible/galaxy/issues/3292

File: roles/add_livemigration_user/README.md
Comment (Revalioli) [reply roles/add_livemigration_user/README.md]: Removed, this was forgot after rebasing on the new debiantcentos branch after merge of https://github.com/seapath/ansible/pull/643

File: roles/ceph_prepare_installation/README.md
Comment (Revalioli) [reply roles/ceph_prepare_installation/README.md]: Same as for `roles/add_livemigration_user`

**Comment by ebail** [comment]
Hi @insatomcat ! Have you got some comments on that ? Thanks

**Review by eroussy — DISMISSED** [review]

**Review by eroussy — DISMISSED** [review]

**Review by eroussy — APPROVED** [review]

**Pull request merged**
